src/main/kotlin/net/lumalyte/lg/infrastructure/services/BankServiceBukkit.kt

```
@@ -366,15 +387,37 @@ class BankServiceBukkit(
-                    guildId = guildId,
-                    actorId = playerId,
-                    action = AuditAction.PERMISSION_DENIED,
-                    details = "Failed to deposit money to player account - withdrawal reverted"
-                ))
+                // Player payout failed AFTER debiting the guild — the re-credit is itself an
+                // external write that can throw or be rejected. If it doesn't succeed the guild
+                // is stranded and the player got nothing; audit it as CRITICAL so the operator
+                // can intervene instead of finding a missing balance days later.
+                val reCreditOk = try {
+                    vaultInventoryManager.depositGold(guildId, playerId, totalDebit.toLong()) >= 0
+                } catch (re: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)